sdk/Pulumi/Provider/Provider.cs

```diff
@@ -19,9 +19,81 @@
 using Microsoft.AspNetCore.Hosting.Server;
 using Microsoft.AspNetCore.Hosting.Server.Features;
 using Microsoft.AspNetCore.Server.Kestrel.Core;
+using System.Reflection.Metadata;
 
 namespace Pulumi.Experimental.Provider
 {
+    public abstract class Parameters { }
```

Comment:
I wasn't sure if it was best to go with this abstract base class and two concrete classes for the different parameter types, or to just have two nullable fields on ParameterizeRequest for each choice. This feels more correct, because `parameters` has to be one or the other, while the two nullable fields _technically_ could both be filled in. But also a bit less obvious to use compared to just two fields. 
For what it's worth Go also does something like this abstract class trick for the parameters.